Automattic/newspack-plugin
# feat: automatically disable guest authors

### All Submissions:






### Changes proposed in this Pull Request:



After the introduction of the Guest Contributor role, we now want to be more proactive and disable CAP's guest authors if the site is not using the feature yet.

CAP's Guest Authors feature will be disabled by default if there are no Guest Authors in the site. If you want to force enabling it, add the NEWSPACK_ENABLE_CAP_GUEST_AUTHORS constant to your wp-config.php file.

### How to test the changes in this Pull Request:

1. In a site that has some Guest Authors
2. Enable this branch and confirm the feature is still enabled and working
3. Check that the option was set: `wp option get newspack_check_site_has_cap_guest_authors` should return `yes`
4. Delete all your guest authors (or start with a new site)
5. Make sure to delete the option: `wp option delete newspack_check_site_has_cap_guest_authors`
6. Load any page of the site again
7. Confirm Guest Authors are disabled and you see our replacement page instead
8. Confirm the option now has the value `no`
9. Add `define( 'NEWSPACK_ENABLE_CAP_GUEST_AUTHORS', true );` to wp-config
10. Confirm CAP's Guest Authors are re-enabled

Testing CRON

1. Repeat steps 1 to 4
2. Run  wp cron event run newspack_re_check_guest_authors
3. Confirm Guest Authors are disabled and you see our replacement page instead
4. Confirm the option now has the value `no`

### Other information:
